src/auto_note.cpp

```diff
@@ -503,41 +501,14 @@ void auto_note_manager_gui::show()
                 global_mapExtraCache )[currentItem];
         const int scroll_rate = cacheSize > 20 ? 10 : 3;
 
-        if( currentAction == "UP" ) {
-            if( currentLine > 0 ) {
-                --currentLine;
-            } else {
-                currentLine = cacheSize - 1;
-            }
-        } else if( currentAction == "DOWN" ) {
-            if( currentLine == cacheSize - 1 ) {
-                currentLine = 0;
-            } else {
-                ++currentLine;
-            }
-        } else if( currentAction == "PAGE_DOWN" ) {
-            if( currentLine == cacheSize - 1 ) {
-                currentLine = 0;
-            } else if( currentLine + scroll_rate >= cacheSize ) {
-                currentLine = cacheSize - 1;
-            } else {
-                currentLine += +scroll_rate;
-            }
-        } else if( currentAction == "PAGE_UP" ) {
-            if( currentLine == 0 ) {
-                currentLine = cacheSize - 1;
-            } else if( currentLine <= scroll_rate ) {
-                currentLine = 0;
-            } else {
-                currentLine += -scroll_rate;
-            }
-        }  else if( currentAction == "ENABLE_MAPEXTRA_NOTE" ) {
+        if( navigate_ui_list( action, currentLine, scroll_rate, cacheSize, true ) ) {
```

Comment:
The replacement above, like the many similar instances later, is the original and primary purpose of this PR.